cognite/powerops/resync/v1/data_classes.py

```diff
@@ -0,0 +1,449 @@
+    overwrite: bool
+    file_metadata: dict[str, FileMetadataWrite]
+
+    def __init__(
+        self,
+        working_directory: Optional[Path] = None,
+        folder_path: Optional[str] = None,
+        overwrite: bool = True,
+        file_metadata: Optional[list[dict[str, Any]]] = None,
+    ) -> None:
+        """Initializes the FileUploadConfiguration with the provided values and processes some of the values."""
+
+        if folder_path:
+            self.folder_path = Path(folder_path)
+        elif working_directory:
+            self.folder_path = working_directory / "files"
+        else:
+            raise ValueError("Either folder_path or working_directory must be provided")
+        self.overwrite = overwrite
+
+        self.file_metadata = {}
+        if file_metadata:
+            for metadata in file_metadata:
+                if "name" not in metadata:
+                    raise ValueError("File metadata must contain a name field")
+                if "directory" in metadata:
+                    metadata["directory"] = self._convert_string_to_unix_path(metadata["directory"])
+
+                file_metadata_instance = FileMetadataWrite(**metadata)
+                if file_metadata_instance.name:
+                    self.file_metadata[file_metadata_instance.name] = file_metadata_instance
+                else:
+                    raise ValueError("File metadata must contain a name field")
+
+    @staticmethod
+    def _convert_string_to_unix_path(path_str: str) -> str:
+        path = Path(path_str).as_posix()
+
+        return path if path.startswith("/") else f"/{path}"
+
+
+@dataclass
+class PropertyConfiguration:
+    """Data class for property configuration.
+
+    Attributes:
+        property: The property name.
+        source_file: The source file to extract the property from.
+        extraction_path: The extraction path in the source file.
+        default_value: The default value if the property is not found.
+        is_subtype: Whether the property is a subtype.
+        is_list: Whether the property is a list.
+        cast_type: The type to cast the property to.
+        cast_type_str: The string representation of the cast type.
+    """
+
+    property: str
+    source_file: Optional[Path]
+    extraction_path: Optional[list[str]]
+    default_value: Optional[Any]
+    is_subtype: bool
+    is_list: bool
+    cast_type: Optional[type]
+    cast_type_str: Optional[str]
+
+    def __init__(
+        self,
+        property: str,
+        source_file: Optional[Path] = None,
+        extraction_path: Optional[str] = None,
+        default_value: Optional[Any] = None,
+        is_subtype: bool = False,
+        is_list: bool = False,
+        cast_type: Optional[str] = None,
+    ) -> None:
+        """Initializes the PropertyConfiguration with the provided values and processes some of the values."""
+
+        self.property = property
+
+        type_mapping = self.generate_string_type_mapping()
```

Comment:
Seems inefficient to _generate_ string type mapping at every instantiation - could we use a static/class variable for this (just add type_mapping = generate_string_type_mapping() to the class, and then refer to PropertyConfiguration.type_mapping where we need it)?

(Maybe no point in doing this kind of optimisation, but should be quite simple)